dotnet-test-samples/schema-and-contract-testing/tests/SchemaTesting.UnitTest/ContractTests.cs

```diff
@@ -0,0 +1,160 @@
+using System.Text.Json;
+using CreateCustomerFunction;
+using Newtonsoft.Json.Linq;
+using Newtonsoft.Json.Schema;
+
+namespace SchemaTesting.UnitTest;
+
+public class ContractTests
+{
+    private readonly List<CustomerCreatedEvent?> _publishedEvents;
+    private readonly Mock<IEventPublisher> _mockEventPublisher;
+    private readonly ISchemaReader _schemaReader;
+
+    public ContractTests()
+    {
+        this._publishedEvents = new List<CustomerCreatedEvent?>();
```

Comment:
Looks like a better solution would be to use automocking container, it would prevent someone accidetly changing the common initialization in a real world scenario